app/models/assessment.rb

```diff
@@ -1,14 +1,31 @@
 # coding: utf-8
 class Assessment < ApplicationRecord
 
-  belongs_to :format
-  belongs_to :created_by, class_name: 'Member'
+  belongs_to :member
+  has_many :questions, inverse_of: :assessment
+  has_many :other_versions,
+           -> (a) { where.not(id: a.id) },
+           class_name: 'Assessment',
+           foreign_key: :identifier, primary_key: :identifier
 
-  enum visibility: %i[internal external]
 
-  has_many :questions, inverse_of: :assessment
-  validates :format, :created_by, presence: true
+  enum visibility: %i[internal external]
 
   validates :version, uniqueness: { scope: :identifier }
```

Comment:
Please add an index in the DB to enforce this constraint there (to prevent race condition uniqueness violations)